### PR TITLE
[web][api][SIMS-#227] Crete User table and add remove user api

### PR DESCRIPTION
**Pull request by fwpushan**
----
1. Create user table to display all users of institution
2. Create api to delete user from institution (Currently using hard delete)
--

### CODE REVIEW

File: sources/packages/api/src/services/institution/institution.service.ts [hunk 1]
```diff
@@ -313,4 +313,13 @@ export class InstitutionService extends RecordDataModelService<Institution> {
         .map((roleObject) => roleObject.role),
     };
   }
+
+  async removeUser(institutionUser: InstitutionUser) {
+    await this.institutionUserRepo.remove(institutionUser);
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/institution/institution.service.ts:318]: Please consider changing both queries to be executed in Parallel.
Please considering decorating the method with a transaction.

File: sources/packages/api/src/route-controllers/institution/institution.controller.ts [hunk 2]
```diff
@@ -167,4 +170,22 @@ export class InstitutionController extends BaseController {
...
     return this.institutionService.getUserTypesAndRoles();
   }
+
+  @Delete("/user/:id")
+  async removeUserAuth(
+    @Param("id") id: string,
+    @UserToken() token: IUserToken,
+  ) {
+    // Get auth
+    const institutionUser = await this.institutionService.getUser(+id);
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution/institution.controller.ts:180]: Is this method ensuring that the user being deleted belongs to the current institution?

File: sources/packages/api/src/route-controllers/institution/institution.controller.ts [hunk 3]
```diff
@@ -167,4 +170,22 @@ export class InstitutionController extends BaseController {
   async getUserTypesAndRoles(): Promise<InstitutionUserTypeAndRoleResponseDto> {
     return this.institutionService.getUserTypesAndRoles();
   }
+
+  @Delete("/user/:id")
+  async removeUserAuth(
+    @Param("id") id: string,
```
[the remark this answers is not in the document]
Comment (fwpushan) [reply sources/packages/api/src/route-controllers/institution/institution.controller.ts:176]: All the example in nestjs is saying @Param() decorator returns string ....

File: sources/packages/api/src/route-controllers/institution/institution.controller.ts (hunk 2 above)
Comment (fwpushan) [reply sources/packages/api/src/route-controllers/institution/institution.controller.ts:180]: 👍

File: sources/packages/api/src/route-controllers/institution/institution.controller.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/institution/institution.controller.ts:176]: It seems that we already have an example working on our API, as shown below.
image

**Pull request merged**